src/olympia/promoted/signals.py

```diff
@@ -67,15 +67,34 @@ def promoted_approval_to_promoted_addon_version(
     signal: ModelSignal,
     instance: PromotedApproval,
 ):
-    if signal == models.signals.post_save:
-        PromotedAddonVersion.objects.update_or_create(
-            version=instance.version,
-            promoted_group=PromotedGroup.objects.get(group_id=instance.group_id),
-            application_id=instance.application_id,
+    # Get all valid approvals for this version
+    valid_approvals = []
+    for approval in PromotedApproval.objects.filter(version=instance.version):
+        # Skip invalid approvals
+        if (
+            approval.group_id is None
+            or approval.group_id == PROMOTED_GROUP_CHOICES.NOT_PROMOTED
+            or approval.application_id is None
```

Comment:
So I:
1. Created a `PromotedApproval` instance with null `application_id`
2. Ran sync_promoted_addons without error

And it does not create any new PromotedAddonVersion instances, as you had written. But I would have expected it to create two (one for each app in `APP_USAGE`)?

Why are we considering application_id=None invalid here, again?

Reply:
AFAICT this is the only place where we create PromotedApproval instances

https://github.com/mozilla/addons-server/blob/d3ed1dddfa1e50d71203eb11b8caa5cbc9a3836e/src/olympia/promoted/models.py#L247-L250

Even though the `PromotedAddon` has a null application, it still returns a list of valid application_id values, which are then saved on the PromotedVersion.

It's not clear to me when or how we would nullify that, so it's actually not clear why that value would be nullable on the PromotedApproval model, but in either case we are going to get a signal for each application_id and we will create one version for each.. if there is a case of a PromotedApproval missing one, this seems like an edge case in the existing code.

Am I missing something?